Lares/Assets/DDOL systems/Inventory/Inventory.cs

```diff
@@ -0,0 +1,160 @@
+using NUnit.Framework;
+using System;
+using UnityEngine;
+using UnityEngine.Experimental.AI;
+using UnityEngine.UI;
+using System.Collections.Generic;
+using static UnityEngine.ParticleSystem;
+using System.Linq;
+
+namespace Lares.Inventory
+{
+    public enum ItemTypes
+    {
+        RECOVERY,
+        MATERIAL,
+        KEY
+    }
+
+    [Serializable]
+    public struct InventoryItem
+    {
+        public int itemID;
+        public string itemName;
+        public string itemDescription;
+        public Sprite itemSprite;
+        public ItemTypes itemType;
+        public onUseFunctions onUse;
+
+        public InventoryItem(int _itemID, string _itemName, string _itemDes, Sprite _itemimage, ItemTypes _itemType, onUseFunctions _onUse)
+        {
+            itemID = _itemID;
+            itemName = _itemName;
+            itemDescription = _itemDes;
+            itemType = _itemType;
+            itemSprite = _itemimage;
+            itemType = _itemType;
+            onUse = _onUse;
+        }
+
+        public static bool operator == (InventoryItem a, InventoryItem b) { return a.itemID == b.itemID; }
+        public static bool operator != (InventoryItem a, InventoryItem b) { return a.itemID != b.itemID; }
+
+        public override bool Equals(System.Object obj)
+        {
+            if (obj == null || !(obj is InventoryItem))
+                return false;
+            else
+                return itemID == ((InventoryItem)obj).itemID;
+        }
+    }
```

Comment:
In this case, you do not need the else, which may cause unnecessary branching and also is just a bit cleaner.

```suggestion
        public override bool Equals(System.Object obj)
        {
            if (obj == null || !(obj is InventoryItem))
                return false;
            
            return ItemId == ((InventoryItem)obj).ItemId;
        }
    }
```